dev/devicelab/bin/tasks/technical_debt__cost.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:path/path.dart' as path;
+
+// the numbers below are odd, so that the totals don't seem round. :-)
+const double todoCost = 1009.0; // about two average SWE days, in dollars
+const double ignoreCost = 2003.0; // four average SWE days, in dollars
+const double pythonCost = 3001.0; //  six average SWE days, in dollars
+
+final RegExp todoPattern = new RegExp(r'// *TODO');
+final RegExp ignorePattern = new RegExp(r'// *ignore:');
+
+Stream<double> findCostsForFile(File file) {
+  if (path.extension(file.path) == '.py')
+    return new Stream<double>.fromIterable(<double>[pythonCost]);
+  if (path.extension(file.path) != '.dart')
+    return null;
+  StreamController<double> result = new StreamController<double>();
+  file.openRead().transform(UTF8.decoder).transform(const LineSplitter()).listen((String line) {
+    if (line.contains(todoPattern))
+      result.add(todoCost);
+    if (line.contains(ignorePattern))
+      result.add(ignoreCost);
+  }, onDone: () { result.close(); });
+  return result.stream;
+}
+
+Stream<double> findCostsForDirectory(Directory directory, Set<String> gitFiles) {
+  StreamController<double> result = new StreamController<double>();
+  Set<StreamSubscription<dynamic>> subscriptions = new Set<StreamSubscription<dynamic>>();
+
+  void checkDone(StreamSubscription<dynamic> subscription, String path) {
+    subscriptions.remove(subscription);
+    if (subscriptions.isEmpty)
+      result.close();
+  }
+
+  StreamSubscription<FileSystemEntity> listSubscription;
+  subscriptions.add(listSubscription = directory.list(followLinks: false).listen((FileSystemEntity entity) {
+    String name = path.relative(entity.path, from: flutterDirectory.path);
+    if (gitFiles.contains(name)) {
+      if (entity is File) {
+        StreamSubscription<double> subscription;
+        subscription = findCostsForFile(entity)?.listen((double cost) {
```

Comment:
I think it can be used inside `findCostsForFile` for iterating over lines and on this line. But you are right, if we use it to iterate over files, it will prevent parallelism across files.